plugins/metrics/pkg/agent/runner.go

```diff
@@ -274,9 +275,21 @@ func (tr *taskRunner) OnTaskRunning(ctx context.Context, activeTask task.ActiveT
 				Timeseries: dereferenceResultTimeseries(result.Timeseries),
 			}
 
-			if err := tr.doPush(ctx, &writeRequest); err != nil {
-				activeTask.AddLogEntry(zapcore.ErrorLevel, err.Error())
-				return err
+			chunkedRequests, err := splitChunksWithLimit(&writeRequest, limit)
+			if err != nil {
+				return fmt.Errorf("failed to chunk request: %w", err)
+			}
+
+			if len(chunkedRequests) > 1 {
+				activeTask.AddLogEntry(zapcore.DebugLevel, fmt.Sprintf("split write request into %d chunks", len(chunkedRequests)))
+			}
+
+			for i, request := range chunkedRequests {
+				if err := tr.doPush(ctx, request); err != nil {
+					activeTask.AddLogEntry(zapcore.ErrorLevel, err.Error())
+					return err
```

Comment:
should there be additional error handling logic here? e.g. handling 429

Reply:
I think the current error handling backoff logic in `doPush` should already be sufficient. We could make that backoff configurable, but I think that level of configuration for the user likely isn't necessary.